src/app/home/home.component.ts

```diff
@@ -41,21 +41,14 @@ export class HomeComponent {
       this.location.go(path);
     }
   }
-  
-  get isCatalogueLoading(): boolean {
-    return this.catalogueService.isLoading;
-  }
 
   get tabIndex(): number {
-    if(this.isCatalogueLoading)
-      return -1;
-
     const tabName = this.route.snapshot.paramMap.get('tab');
```

Comment:
you could use the route service as dependency of catalogue service and move the entire method into catalogueService

and use catalogueService.tabIndex in the web html view

Reply:
i tried, but the route dependency didn't work, maybe only works only on component class :thinking:

Reply:
ok may it's not a good practice but you could pass "this.router" to the method of the service

I got this idea from https://stackoverflow.com/questions/40219790/angular-2-get-routeparams-in-a-service

but I am still not convinced

we could even leave as it is

Reply:
solved